kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/metadata.kt

```diff
@@ -374,6 +375,25 @@ private val TargetProperty.isVisible: Boolean
         || visibility == KModifier.PUBLIC
   }
 
+private val AnnotationSpec.isTransient: Boolean
+  get() {
+    if (className == Transient::class.asClassName()) {
+      return true
+    }
+
+    if (className == JSON) {
+      val mirror = requireNotNull(tag<AnnotationMirror>()) {
+        "Could not get the annotation mirror from the annotation spec"
+      }
+      val isTransient = mirror.elementValues.entries.singleOrNull {
+        it.key.simpleName.contentEquals("isTransient")
+      } ?: return false
```

Comment:
Let's make a reusable function here rather than duplicate the logic from the `jsonName()` function you pulled this from and consolidate

```kotlin
private operator fun Map<out ExecutableElement, AnnotationValue>.get(member: String): Any? {
  return entries.firstOrNull {
    it.key.simpleName.contentEquals(member)
  }?.value?.value
}

// Then you can do this!
val isTransient = mirror.elementValues["isTransient"] as? Boolean ?: false
```